xlunch.c

```diff
@@ -1078,33 +1083,34 @@ void run_command(char * cmd_orig)
     if (dont_quit)
     {
         pid_t pid=fork();
-        if (pid==0) // child process
-        {
+
+        switch (pid) {
+        case 0:   /* Child process */
             if (!multiple_instances) close(lock);
             printf("Forking command: %s\n",cmd);
-            int err;
-            err = execvp(array[0],array);
-            fprintf(stderr,"Error forking %s : %d\n",cmd,err);
-            exit(OKAY);
-        }
-        else if (pid<0) // error forking
-        {
-            fprintf(stderr,"Error running %s\n",cmd);
-        }
-        else // parent process
-        {
+            break;
+
+        case -1:  /* Error */
+            perror("fork");
```

Comment:
This should include a more informative message than just the word "fork". Probably the best is to read `errno` and convert it to an error message.

Reply:
`perror(3)` does this for you. `perror(str)` is essentially
```
    fprintf(stderr, "%s: %s\n", str, strerror(errno));
```

Reply:
Ah I see, should be fine then